### PR TITLE
Adding support for scenarios where the RMI registry has SSL enabled

### PR DESCRIPTION
**Pull request by dmitchnewrelic**
**Description:**

The JMX Metric Gatherer is not working properly with servers that have SSL protection enabled on the RMI registry.  [Oracle's Java documentation](https://docs.oracle.com/javadb/10.10.1.2/adminguide/radminjmxenablepwdssl.html) describes this setting as follows: 

_The system property `com.sun.management.jmxremote.registry.ssl=true` was new in JDK 6 and aims at resolving security issues with the RMI registry used in relation with JMX. This property must be used in conjunction with `com.sun.management.jmxremote.ssl.need.client.auth=true` in order to fully secure the RMI registry._

Changes were made to the JMX Metrics Gatherer to support this scenario, by adding a new helper class called JmxConnectorHelper. This class implements different connection logic when SSL is enabled on the RMI registry. 

Changes were also made to JmxConfig, as a new parameter is required to handle this scenario.  Specifically, the following parameter was added: 

```
otel.jmx.remote.registry.ssl=true
```
The readme file was updated to explain when to use this new parameter. 

**Existing Issue(s):**

Issue #834 

**Testing:**

The following scenarios were tested: 

- SSL not enabled, and no credentials required
- SSL not enabled, but with credentials
- SSL enabled for JMX, but not on the RMI registry 
- SSL enabled for JMX including the RMI registry  

**Documentation:**

README.md was updated to reflect the new configuration parameters. 

**Outstanding items:**

Nothing at this time.

### CODE REVIEW

File: jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConnectorHelper.java
Comment (breedx-splk) [inline jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConnectorHelper.java:76]: nit: it's a little clunky to have this method just do side effect. Prefer to return the thing and assign in the caller.

```suggestion
  private static RMIServer getStub(String hostName, int port) throws IOException {
    try {
      Registry registry = LocateRegistry.getRegistry(hostName, port, sslRMIClientSocketFactory);
      return (RMIServer) registry.lookup("jmxrmi");
    } catch (NotBoundException nbe) {
      throw new IOException(nbe);
    }
  }
```

**Review by breedx-splk — APPROVED** [review]
This seems good to me. Had one rather small suggestion. 

I'd really like to see an integration test that covers this. Would you mind opening an issue to do that work in a follow-up PR?

File: jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConnectorHelper.java
Comment (jeffalder) [reply jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConnectorHelper.java:76]: Yo @breedx-splk! Good idea, me gusta

File: jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConnectorHelper.java
Comment (jeffalder) [inline jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConnectorHelper.java:62]: @dmitchnewrelic You'll need to assign the variable here, now.

**Comment by dmitchnewrelic** [comment]
Thanks @breedx-splk.  I've made the suggested change and will create a separate issue for the integration test.

**Comment by dmitchnewrelic** [comment]
I've created issue #838 for this.

**Comment by jeffalder** [comment]
@breedx-splk per https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/RELEASING.md#release-cadence does this mean the next normal release would be May 12th?

**Comment by trask** [comment]
just a heads up, you can ignore the EasyCLA failure, it is having an issue currently (https://github.com/open-telemetry/community/issues/1445)

> @breedx-splk per https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/RELEASING.md#release-cadence does this mean the next normal release would be May 12th?

👍

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by breedx-splk** [comment]
> @breedx-splk per https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/RELEASING.md#release-cadence does this mean the next normal release would be May 12th?

I don't think we'd want to promise a date, but yeah, basically monthly and more strongly coordinated with the core and instrumentation releases. I defer to the maintainers on anything more specific.

**Pull request merged**
